examples/check-semantics/check_semantics.md

```diff
@@ -1,13 +1,11 @@
 # Check semantics
 
 `check` semantics is not to simply return on an `error` value.
-When `check` gets an `error` value, it fails.
-The enclosing block decides how to handle the failure.
-Most blocks pass the failure up to the enclosing block.
-Function definition handles the failure by returning the error.
+When `check` gets an `error` value, it fails. The enclosing block decides how to handle the failure.
+Most blocks pass the failure up to the enclosing block. Function definition handles the failure by returning the error.
 `on fail` can catch the error.
-`fail` statement is like `check`, but it always fails.
-Differs from exceptions in that control flow is explicit.
+
+`fail` statement is like `check`, but it always fails and differs from exceptions in that control flow is explicit.
```

Comment:
```suggestion
The `fail` statement is like `check`, but it always fails. It differs from the exceptions in that the control flow, which are explicit.
```
Please check if the grammatical changes suggested are affecting the technicalities.